adapter/src/dbt/adapters/fal_experimental/utils/environments.py

```diff
@@ -31,17 +35,27 @@ def run(self, executable: BasicCallable, *args, **kwargs) -> Any:
 
 
 def fetch_environment(
-    project_root: str, environment_name: str
+        project_root: str,
+        environment_name: str,
+        machine_type: str = "S",
+        credentials: Optional[Any] = None
 ) -> Tuple[BaseEnvironment, bool]:
     """Fetch the environment with the given name from the project's
     fal_project.yml file."""
     # Local is a special environment where it doesn't need to be defined
     # since it will mirror user's execution context directly.
     if environment_name == "local":
-        return LocalPythonEnvironment(), True
+        if credentials.key_secret and credentials.key_id:
+            # "local" environment in this case is fal cloud
```

Comment:
So user can say `environment: local` and it will run in the cloud? Maybe we want to show a warning or change this?

Reply:
Warning is probably better. We could ask users to use a profile target without a cloud definition to run things locally.